File: app/src/main/java/com/commit451/pixeladjustor/sample/MainActivity.java
package com.commit451.pixeladjustor.sample;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.graphics.Color;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.widget.ImageView;

import com.commit451.pixeladjuster.PixelAdjuster;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        BitmapFactory.Options opts = new BitmapFactory.Options();
        opts.inMutable = true;
        Bitmap image = BitmapFactory.decodeResource(getResources(), R.drawable.horizon, opts);

        ImageView imageView = (ImageView) findViewById(R.id.image);

        long startTime = System.currentTimeMillis();
        JavaPixelAdjuster.adjustColor(image, Color.WHITE, Color.MAGENTA);

        Log.d("TIMER", "Java Time taken: " + (System.currentTimeMillis() - startTime) + " ms");
        imageView.setImageBitmap(image);

        ImageView nativeImageView = (ImageView) findViewById(R.id.natively_processed_image);

        Bitmap image2 = BitmapFactory.decodeResource(getResources(), R.drawable.horizon, opts);

        startTime = System.currentTimeMillis();
        PixelAdjuster.adjustColor(image2, Color.WHITE, Color.MAGENTA);
        Log.d("TIMER", "Native Time taken: " + (System.currentTimeMillis() - startTime) + " ms");
        nativeImageView.setImageBitmap(image2);
    }




}

File: app/src/main/java/com/commit451/pixeladjustor/sample/MainActivity.kt
package com.commit451.pixeladjustor.sample

import android.graphics.BitmapFactory
import android.graphics.Color
import android.os.Bundle

import com.commit451.pixeladjuster.PixelAdjuster

import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val opts = BitmapFactory.Options()
        opts.inMutable = true
        val image = BitmapFactory.decodeResource(resources, R.drawable.horizon, opts)

        var startTime = System.currentTimeMillis()
        KotlinPixelAdjuster.adjustColor(image, Color.WHITE, Color.MAGENTA)

        textKotlin.text = "Kotlin Time taken: ${(System.currentTimeMillis() - startTime)} ms"
        imageKotlin.setImageBitmap(image)

        val image2 = BitmapFactory.decodeResource(resources, R.drawable.horizon, opts)

        startTime = System.currentTimeMillis()
        PixelAdjuster.adjustColor(image2, Color.WHITE, Color.MAGENTA)
        textNdk.text = "NDK Time taken: ${(System.currentTimeMillis() - startTime)} ms"
        imageNdk.setImageBitmap(image2)
    }


}
